enhancements/project-request-limit.md

```diff
@@ -0,0 +1,392 @@
+spec:
+  group: operator.project.openshift.io
+  version: v1
+  scope: Cluster
+  names:
+    plural: projectrequestlimits
+    singular: projectrequestlimit
+    kind: ProjectRequestLimit
+```  
+
+This is how a cluster-admin can enable the `ProjectRequestLimit` admission webhook on a cluster with `limits` to be enforced. 
+```
+apiVersion: operator.project.openshift.io/v1
+kind: ProjectRequestLimit
+metadata:
+  name: cluster
+spec:
+  limits:
+  // for selector level=admin, no maxProjects is specified. This means that users with this label 
+  // will not have a maximum of project requests.
+  - selector:
+      level: admin
+  
+  //  for selector level=advanced, a maximum number of 10 projects will be allowed.
+  - selector:
+      level: advanced
+    maxProjects: 10
+  
+  // no selector is specified. This means that it will be applied to any user that doesn’t satisfy 
+  // the previous two rules. Because rules are evaluated in order, this rule should be specified last.
+  - maxProjects: 2
+```
+
+##### Install:
+* The operator is installed into a predefined namespace by OLM: The predefined namespace `openshift-project-request-limit-operator` 
+  will be wired in to the OLM manifests.
+* The operator will install the `ProjectRequestLimit` admission webhook into the same namespace as the operator. 
+
+
+##### Configuration:
+The operand is immutable, it loads the configuration from a file when it starts. A `ConfigMap` containing the configuration
+is mounted as a volume into the Pod at a certain file path.
+```
+apiVersion: apps/v1
+kind: DaemonSet
+spec:
+  template:
+    metadata:
+      annotations:
+        projectrequestlimits.operator.project.openshift.io/configuration.hash: abcdefgh
+    spec:
+      containers:
+        - name: project-request-limit-admission
+          command:
+            - /usr/bin/project-request-limit-admission
+          args:
+            - "--configuration=$(CONFIGURATION_PATH)"
+          env:
+            - name: CONFIGURATION_PATH
+              value: /etc/projectrequestlimit/config/limits.yaml
+          volumeMounts:
+            name: configuration
+            mountPath: /etc/projectrequestlimit/config/limits.yaml
+            subPath: limits.yaml
+      volumes:
+        - name: configuration
+          configMap:
+            name: project-request-limit-configuration
+``` 
+
+The operator will ensure that change(s) made to the configuration in the `cluster` custom resource are propagated to the 
+`ProjectRequestLimit` admission webhook immediately.
+* The cluster admin changes the configuration inside the `cluster` custom resource.
+* The operator detects the change and compares the hash of the `desired` configuration with that of the `current` configuration.
+* If the configuration has changed, the operator updates the `ConfigMap` object and will set the annotation 
+  `projectrequestlimits.operator.project.openshift.io/configuration.hash` of the template `PodSpec` of the operand.
+
```

Comment:
note that the configmap is change live in the pod by the kubelet. You could also react to changes directly. That's a similar discussion as watching the CR.

Reply:
yes, and I think as a goal we want to keep the runtime complexity of the admission webhook to a minimum. So we let the operator to manage it, this allows us to confine all of operand's lifecycle management to the operator logic.